src/nvme/ioctl.c

```diff
@@ -525,7 +525,7 @@ int nvme_get_log_page(int fd, __u32 xfer_len, struct nvme_get_log_args *args)
 int nvme_set_features(struct nvme_set_features_args *args)
 {
 	__u32 cdw10 = NVME_SET(args->fid, FEATURES_CDW10_FID) |
-			NVME_SET(!!args->save, SET_FEATURES_CDW10_SAVE);
+			NVME_SET((__u32)!!args->save, SET_FEATURES_CDW10_SAVE);
```

Comment:
Wouldn't it make sense to fix the `NVME_SET` macro to handle this correctly? I really do not like sprinling around  casts in the code base.

Reply:
That's a good idea. is `NVME_SET()` assumed to always operate on `u32` values?

Reply:
Not sure. Maybe we can peek into the Linux kernel see how the handle this. I am sure we are not the first one running into this issue :)